### PR TITLE
Add documentation for gamm module

### PR DESCRIPTION
**Pull request by mattverse**
This pull request adds the documentation for gamm module, specifically the concepts of following.

- Pool and share
- Swap
- Multihop

### CODE REVIEW

File: x/gamm/spec/01_concepts.md [hunk 1]
```diff
@@ -4,5 +4,50 @@ order: 1
...
+
+## Pool
+
+### Creation of Pool
+
+At an initial creation of the pool, a fixed amount of 100 share token is minted in the pool and sent to the creator of the pool's account. Pool share denom is in the format of gamm/pool/{poolId} and is displayed in the format of GAMM-{poolId} to the user. Pool assets are sorted in alphabetical order by defualt.
+
+### Joining Pool
+
+When joining a pool, users provide maximum amount of tokens willing to deposit, while the FE takes care of the calculation of how many share tokens the user is eligible at the specific moment of sending the transaction. Calculation of exactly how many tokens are needed to get the designated share is done at the moment of procssing the transaction, validating that it does not exceed the maximum amount of token the user is willing to deposit. After the validation, the share of the pool is minted and sent to the user account. Joining the pool using a single asset is also possible.
```
Comment (ValarDragon) [inline x/gamm/spec/01_concepts.md:17]: FE?

Comment (mattverse) [reply x/gamm/spec/01_concepts.md:17]: Oh I meant Front End in the context that the JoinPool function takes in how much share token as a parameter. From what I know, the calculation is done in the front-end, providing the function with the amount of share token the user is eligible for, while the user only provides the max-token. Do I have a misunderstanding somewhere?

Comment (sunnya97) [reply x/gamm/spec/01_concepts.md:17]: I think @ValarDragon just didn't realize what FE meant.  Probably just write out "frontend"

Comment (mattverse) [reply x/gamm/spec/01_concepts.md]: Ah I see, fixed!

**Review by ValarDragon — APPROVED** [review]

**Comment by ValarDragon** [comment]
Ty for making this! Sorry for the delay in merge.

Theres some more docs we can add, but absolutely no reason to block merge of this!

**Pull request merged**